### ISSUE TITLE
proposal: extend object-curly-newline to allow single line and single prop in new line

### ISSUE THREAD

**Issue by kentor** [eslint/eslint#7382]
**What rule do you want to change?**
[object-curly-newline](http://eslint.org/docs/rules/object-curly-newline)

**Does this change cause the rule to produce more or fewer warnings?**
fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option

**Please provide some example code that this change will affect:**

``` js
// pattern 1
var obj = {
  foo: 'bar',
};

// pattern 2
var obj = { foo: 'bar' };

// pattern 3
var obj = { foo: 'bar',
            baz: 'test' }
```

**What does the rule currently do for this code?**
Not warn for pattern 2. Warns for pattern 1 and 3. 

**What will the rule do after it's changed?**
Not warn for pattern 1 and 2. Still warns for 3.

**Tell us about your environment**
- **ESLint Version:** 3.8
- **Node Version:** 6.8
- **npm Version:** 3.10

**What parser (default, Babel-ESLint, etc.) are you using?** default

**more info:**

The current `{ multiline: true }` option (the default) has the behavior described above.

pattern 1 is useful if the dev thinks more props will be added.
pattern 2 is fine if the dev thinks no more props will be added.
pattern 3 is bad for diffs if more props are added

I would like an option to prevent 3, but not be so strict as to not allow 1.

**Comment by vitorbal** [comment]
@kentor thank you for your issue. Could you please take the time to fill in the [template for proposing rule changes](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md)?
This will help the team understand what you are proposing and allow us to give quicker feedback on your proposal. Thanks a lot!

**Comment by eslintbot** [comment]
Hi @kentor, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by kentor** [comment]
@vitorbal thanks, updated.

**Comment by platinumazure** [comment]
Endorsed (assuming we can come up with a good option name).

**Comment by vitorbal** [comment]
@kentor thanks for the quick reply! I think this makes sense. How would you see this new option being named?
One more thing. Would you be willing to work on a PR for this? Currently, the team is very focused on improving compatibility between JSCS and ESLint, so if you are willing to work on a PR for this, there's a way higher chance that this could end up being accepted.

**Comment by kentor** [comment]
Sure I can work on a PR if we decide on a name.

The naming is a bit tricky. To me `{ multiline: true }` is a good name, except right now it means "if there are line breaks between multiple props" instead of "if the object expression takes up multiple lines".

It would be nice if the default is renamed to `{ multiprop-multiline: true }` but would break backwards compat.

maybe it could be named `object-multiline`

**Comment by kaicataldo** [comment]
Closing this as discussion has stalled and it doesn't appear there is enough interest, either from the team or from the community, to continue the discussion at this time. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. Please feel free to open a new issue with a proposal to continue this discussion if this is something you would like to continue advocating for.
